src/index.ts

```diff
@@ -161,18 +161,46 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
       // so remove the header from the response we'll return
       headers.delete('cache-control')
     }
-    response = new Response(response.body, { headers })
+
+    let shouldRevalidate = false
+    // Several preconditions must be met for a 304 Not Modified:
+    // - mime type cannot be HTML. there several reasons for this
+    // but the most glaring is that CF uses chunked transfer encoding
+    // when inserting html via cache api. Thus the body is streamed
+    // before the full file contents are known
```

Comment:
I'm curious about this. In the call to cache.put() later on in this file, we use a response body which is backed by an ArrayBuffer, so I would expect the runtime to PUT a Content-Length-ful response to cache. That's not what you see happen?

If there's a possibility that we could relax this non-HTML restriction, that'd be ideal, since then we could send the `if-none-match` header in the `cache.match()` request, and have the cache return any 304 directly.

Reply:
Thank you both for this. Sorry to butt in...

FWIW, I have a modified version of KV Asset Mapper for a hobby/experiment project and _I think_ I have .html behaving the same as all other assets copying almost exactly what shagamemnon has done. 304 on HTML should be okay... Either way agree that this is worth to keep exploring/test to turbo charge those load times

Reply:
Thanks for the comment! Everyone's welcome to butt in. :)

I had come into this PR a bit cold, but I've since read the linked issue, and the rest of this file. It seems that the code comment here refers to the difficulty of generating a hash of gigantic body without going over CPU limits, but the caching strategy was later changed to use the existing content hash from the wrangler upload process. It also looks like if we're executing in this part of the code, then we are using wrangler's generated content hash (because `response` will only be truthy if `shouldEdgeCache` is true).

So, to my eyes, it should already be safe to send the conditional request to cache and have cache return the 304.

Reply:
This was an oversight on my part. On an earlier commit, I was terminating function execution early (https://github.com/cloudflare/kv-asset-handler/pull/94/commits/13faf6f195407aa2efd3e8fe760004665f388d85#diff-f41e9d04a45c83f3b6f6e630f10117feR182-R186) and mistakenly that `Transfer-Encoding: chunked` was imposing a limitation on HTML. Barring any unforeseen side-effects, I will remove the HTML restriction